src/test/java/com/clickhouse/kafka/connect/sink/ClickHouseSinkTaskWithSchemaTest.java

```diff
@@ -341,6 +389,33 @@ public void specialCharTableNameTest() {
         assertEquals(sr.size(), countRows(chc, topic));
     }
 
+    @Test
+    // https://github.com/ClickHouse/clickhouse-kafka-connect/issues/57
+    public void supportDatesTest() {
+        Map<String, String> props = new HashMap<>();
+        props.put(ClickHouseSinkConnector.HOSTNAME, db.getHost());
+        props.put(ClickHouseSinkConnector.PORT, db.getFirstMappedPort().toString());
+        props.put(ClickHouseSinkConnector.DATABASE, "default");
+        props.put(ClickHouseSinkConnector.USERNAME, db.getUsername());
+        props.put(ClickHouseSinkConnector.PASSWORD, db.getPassword());
+        props.put(ClickHouseSinkConnector.SSL_ENABLED, "false");
+
+        ClickHouseHelperClient chc = createClient(props);
+
+        String topic = "support-dates-table-test";
+        dropTable(chc, topic);
+        // , date_number Date, date32_number Date32 , datetime_number DateTime
+        createTable(chc, topic, "CREATE TABLE `%s` ( `off16` Int16, date_number Date, date32_number Date32, datetime_number DateTime, datetime64_number DateTime64 ) Engine = MergeTree ORDER BY off16");
+        // https://github.com/apache/kafka/blob/trunk/connect/api/src/test/java/org/apache/kafka/connect/data/StructTest.java#L95-L98
```

Comment:
Consider using a commit hash instead of a branch name. With the branch name, the content on the select range will diverge in time.